op-challenger/fault/agent.go

```diff
@@ -29,7 +29,10 @@ func NewAgent(loader Loader, maxDepth int, trace TraceProvider, responder Respon
 }
 
 // Act iterates the game & performs all of the next actions.
-func (a *Agent) Act() error {
+func (a *Agent) Act(ctx context.Context) error {
+	if a.tryResolve(ctx) {
+		return nil
+	}
 	game, err := a.newGameFromContracts(context.Background())
```

Comment:
any reason we don't pass thru the `ctx` that is passed as an arg here and instead make a new background context?

Reply:
Oops, good catch!